src/include/sof/lib/dai-zephyr.h

```diff
@@ -0,0 +1,268 @@
+#include <stddef.h>
+#include <stdint.h>
+#include <device.h>
+
+/** \brief If the device does not exist it will be created */
+#define DAI_CREAT		BIT(0)
+
+struct dai {
+	uint32_t index;
+	uint32_t type;
+	uint32_t dma_caps;
+	uint32_t dma_dev;
+	const struct device *drv;
+};
+
+struct timestamp_cfg {
+	uint32_t walclk_rate; /* Rate in Hz, e.g. 19200000 */
+	int type; /* SSP, DMIC, HDA, etc. */
+	int direction; /* Playback, capture */
+	int index; /* For SSPx to select correct timestamp register */
+	int dma_id; /* GPDMA id*/
+	int dma_chan_index; /* Used GPDMA channel */
+	int dma_chan_count; /* Channels in single GPDMA */
+};
+
+struct timestamp_data {
+	uint64_t walclk; /* Wall clock */
+	uint64_t sample; /* Sample count */
+	uint32_t walclk_rate; /* Rate in Hz, e.g. 19200000 */
+};
+
+struct timestamp_ops {
+	int (*ts_config)(struct dai *dai, struct timestamp_cfg *cfg);
+	int (*ts_start)(struct dai *dai, struct timestamp_cfg *cfg);
+	int (*ts_stop)(struct dai *dai, struct timestamp_cfg *cfg);
+	int (*ts_get)(struct dai *dai, struct timestamp_cfg *cfg,
+		      struct timestamp_data *tsd);
+};
+
+/**
+ * \brief DAI group information
+ */
+struct dai_group {
+	/**
+	 * Group ID
+	 */
+	uint32_t group_id;
+
+	/**
+	 * Number of DAIs in this group
+	 */
+	uint32_t num_dais;
+
+	/**
+	 * Number of DAIs to receive a trigger before processing begins
+	 */
+	uint32_t trigger_counter;
+
+	/**
+	 * Trigger command to propagate
+	 */
+	int trigger_cmd;
+
+	/**
+	 * Last trigger error
+	 */
+	int trigger_ret;
+
+	/**
+	 * Group list
+	 */
+	struct list_item list;
+};
+
+/**
+ * \brief DAI runtime data
+ */
+struct dai_data {
+	/* local DMA config */
+	struct dma_chan_data *chan;
+	uint32_t stream_id;
+	struct dma_sg_config config;
+	struct comp_buffer *dma_buffer;
+	struct comp_buffer *local_buffer;
+	struct timestamp_cfg ts_config;
+	struct dai *dai;
+	struct dma *dma;
+	struct dai_group *group;	/**< NULL if no group assigned */
+	int xrun;		/* true if we are doing xrun recovery */
+
+	pcm_converter_func process;	/* processing function */
+
+	uint32_t dai_pos_blks;	/* position in bytes (nearest block) */
+	uint64_t start_position;	/* position on start */
+	uint32_t period_bytes;	/**< number of bytes per one period */
+
+	/* host can read back this value without IPC */
+	uint64_t *dai_pos;
+
+	struct ipc_config_dai ipc_config;	/* generic common config */
+	void *dai_spec_config;	/* dai specific config from the host */
+
+	uint64_t wallclock;	/* wall clock at stream start */
+
+	/*
+	 * flag indicating two-step stop/pause for DAI comp and DAI DMA.
+	 * DAI stop occurs during STREAM_TRIG_STOP IPC and DMA stop during DAI_CONFIG IPC with
+	 * the SOF_DAI_CONFIG_FLAGS_HW_FREE flag.
+	 * DAI pause occurs during STREAM_TRIG_PAUSE IPC and DMA pause during DAI_CONFIG IPC with
+	 * the SOF_DAI_CONFIG_FLAGS_PAUSE flag.
+	 */
+	bool delayed_dma_stop;
+};
+
+/**
+ * \brief Array of DAIs grouped by type.
+ */
+struct dai_type_info {
+	uint32_t type;		/**< Type */
```

Comment:
seems like it could be just an `int` - but you'd need to change that around, e.g. in function arguments too